### PR TITLE
Fix some typos in strings and comments

### PR DESCRIPTION
**Pull request by sndth**
This PR fixes several typos found in a few strings as well as in many comments, but it does not affect the code's functionality in any way. There are many more typos, and I will try to fix some more.

---

## Pull Request Checklist

Read the [CONTRIBUTING.md](https://github.com/fktn-k/fkYAML/blob/develop/CONTRIBUTING.md) file for detailed information.  

- [x] Changes are described in the pull request or in a referenced [issue](https://github.com/fktn-k/fkYAML/issues).
- [x] The test suite compiles and runs without any error.
- [x] [The code coverage](https://coveralls.io/github/fktn-k/fkYAML) on your branch is 100%.
- [x] The documentation is updated if you added/changed a feature.

## Please don't

- The C++11 support varies between different **compilers** and versions. Please note the [list of supported compilers](https://github.com/fktn-k/fkYAML/blob/develop/README.md#supported-compilers). Some compilers like GCC 4.7 (and earlier), Clang 3.3 (and earlier), or Microsoft Visual Studio 13.0 and earlier are known not to work due to missing or incomplete C++11 support. Please refrain from proposing changes that work around these compiler's limitations with `#ifdef`s or other means.
- Please refrain from proposing changes that would **break [YAML](https://yaml.org/) specifications**. If you propose a conformant extension of YAML to be supported by the library, please motivate this extension.
- Please do not open pull requests that address **multiple issues**.

### CODE REVIEW

File: include/fkYAML/detail/conversions/to_string.hpp [hunk 1]
```diff
@@ -6,8 +6,8 @@
 // SPDX-FileCopyrightText: 2023-2024 Kensuke Fukutani <fktn.dev@gmail.com>
 // SPDX-License-Identifier: MIT
 
-#ifndef TO__string_HPP_
-#define TO__string_HPP_
+#ifndef TO_STRING_HPP
+#define TO_STRING_HPP
```
Comment (fktn-k) [inline include/fkYAML/detail/conversions/to_string.hpp:10]: @sndth   
Could you replace this define with `FK_YAML_DETAIL_CONVERSIONS_TO_STRING_HPP` so it matches the other ones?

**Review by fktn-k — CHANGES_REQUESTED** [review]
@sndth   
Thanks for fixing so many typos and gramatical mistakes.  
I requested a tiny change, but the others look good to me.

File: include/fkYAML/detail/conversions/to_string.hpp (hunk 1 above)
Comment (sndth) [reply include/fkYAML/detail/conversions/to_string.hpp]: Of course, done!

**Review by fktn-k — APPROVED** [review]
Looks good to me.

**Comment by fktn-k** [comment]
@sndth   
Let me know when you're done with all fixes.

**Comment by sndth** [comment]
Currently that's all!

**Comment by fktn-k** [comment]
@sndth   
OK. Thanks a lot!  
I'll merge the changes.

**Pull request merged**